aviary/mission/flops_based/phases/two_dof_phase.py

```diff
@@ -0,0 +1,267 @@
+        phase.add_timeseries_output("mass")
+        phase.add_timeseries_output(Dynamic.Mission.ALTITUDE)
+        phase.add_timeseries_output("alpha")
+        phase.add_timeseries_output(Dynamic.Mission.FLIGHT_PATH_ANGLE)
+        phase.add_timeseries_output(Dynamic.Mission.THROTTLE)
+        phase.add_timeseries_output(
+            "fuselage_pitch", output_name="theta", units="deg")
+
+        return phase
+
+    def make_default_transcription(self):
+        '''
+        Return a transcription object to be used by default in build_phase.
+        '''
+        user_options = self.user_options
+
+        num_segments, _ = user_options.get_item('num_segments')
+        order, _ = user_options.get_item('order')
+
+        seg_ends, _ = dm.utils.lgl.lgl(num_segments + 1)
+
+        transcription = dm.Radau(
+            num_segments=num_segments, order=order, compressed=True,
+            segment_ends=seg_ends)
+
+        return transcription
+
+    def _extra_ode_init_kwargs(self):
+        """
+        Return extra kwargs required for initializing the ODE.
+        """
+        # TODO: support external_subsystems and meta_data in the base class
+        return {
+            'external_subsystems': self.external_subsystems,
+            'meta_data': self.meta_data,
+            'subsystem_options': self.subsystem_options,
+            'input_speed_type': SpeedType.MACH,
+            'clean': self.user_options.get_val('clean'),
+            'ground_roll': self.user_options.get_val('ground_roll'),
+            'balance_throttle': self.user_options.get_val('balance_throttle'),
+        }
+
+
+TwoDOFPhase._add_meta_data(
+    'num_segments', val=5, desc='transcription: number of segments')
+
+TwoDOFPhase._add_meta_data(
+    'order', val=3,
+    desc='transcription: order of the state transcription; the order of the control'
+    ' transcription is `order - 1`')
+
+TwoDOFPhase._add_meta_data('fix_initial', val=False)
+
+TwoDOFPhase._add_meta_data('fix_duration', val=False)
+
+TwoDOFPhase._add_meta_data('optimize_mach', val=False)
+
+TwoDOFPhase._add_meta_data('optimize_altitude', val=False)
+
+TwoDOFPhase._add_meta_data('throttle_setting', val=None, desc='throttle setting')
+TwoDOFPhase._add_meta_data('initial_bounds', val=(0., 100.),
+                           units='s', desc='initial bounds')
+TwoDOFPhase._add_meta_data('duration_bounds', val=(
+    0., 3600.), units='s', desc='duration bounds')
+TwoDOFPhase._add_meta_data('initial_ref', val=100., units='s', desc='initial reference')
+TwoDOFPhase._add_meta_data('duration_ref', val=1000.,
+                           units='s', desc='duration reference')
+TwoDOFPhase._add_meta_data('control_order', val=1, desc='control order')
+TwoDOFPhase._add_meta_data('opt', val=True, desc='opt')
+TwoDOFPhase._add_meta_data('ground_roll', val=True)
+TwoDOFPhase._add_meta_data('rotation', val=False)
+TwoDOFPhase._add_meta_data('clean', val=False)
+TwoDOFPhase._add_meta_data('balance_throttle', val=False)
```

Comment:
Should the default be True? I notice that we only set it to False in the phases up to the (obstacle?), but the D-E mic phases don't set it.

Reply:
Great catch, thanks Ken, changed default to False

Reply:
For `balance_throttle` I think the default right now should be False. This means that the user is supplying the throttle value. I'm working on a separate branch right now where it will default to True with a bit more protections in how the throttle is computed; either solved or with path constraints.